js/renovation/dist/button.j.jsx

```diff
@@ -10,22 +14,30 @@ class Button extends Widget {
 
     getProps(isFirstRender) {
         const props = super.getProps(isFirstRender);
-        if(props.contentRender) {
-            props.contentRender = (data) => {
-                const template = this._getTemplate(props.contentRender);
-
-                return (<div style={{ display: 'none' }} ref={(element) => {
-                    if(element && element.parentElement) {
-                        const parent = element.parentElement;
-                        while(parent.firstChild) {
-                            parent.removeChild(parent.firstChild);
-                        }
-                        template.render({ model: data, container: parent });
-                        parent.appendChild(element);
-                    }
-                }}/>);
+
+        if(props.template) {
+            props.contentRender = ({ text, icon, contentRef, children }) => {
+                const data = { text, icon };
+                let $content = $('<div>');
+                $content.addClass('dx-button-content');
```

Comment:
We expect this file to be generated< so it can not contain any control specific content like class names (dx-button-content) etc.
Is there any way to move it somewhere or get it from declarative component?

Reply:
We can move them into properties:
```jsx
props.contentRender = ({ text, icon, className, wrapperClassName, contentRef, children }) => {
    // ...
    $content.addClass(className);
    // ...
    if ($template.hasClass(wrapperClassName)) { }
    // ...
}
```